### ISSUE TITLE
[Bug Report][3.3.14] Global Defaults Passing Through Modals

### ISSUE THREAD

**Issue by Winter979** [vuetifyjs/vuetify#18123]
### Environment
**Vuetify Version:** 3.3.14
**Last working version:** 3.3.13
**Vue Version:** 3.3.4
**Browsers:** Firefox 116.0
**OS:** Mac OS 10.15

### Steps to reproduce
1. Set the Vuetify defaults to
```
  defaults: {
      VTextField: { variant: 'outlined' },
      VTable: { VTextField: { variant: 'plain' } },
  },
```
2. Create a component structure as such
```
|-- VTable
      | -- VTextField (Expected variant=plain)
      | -- VDialog
         |--  VTextField (Expected variant=outlined)
```

### Expected Behavior
The VTextField that is in a dialog located in the table should be outlined

### Actual Behavior
The VTextField has the variant out plain instead of outlined

### Comments
I have realised though that if i was to wrap the VTable in a VDefaultsProvider with an empty defaults prop. then the defaults act as expected.

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqtVktT2zAQ/iuqewidwTEtPbSZwMBQOj11emDggDko1garyJJH2hiYTP57V36E2LFJWnpJom9f2scX7e0yKBaAcv48/u2CSSCz3FhkS5ZY4AjXlYyt2NyajI1q3VGsYw1PpWpitENWC9hJ2/BgGWvGMIUMJuRUwJwvFF5V55GS9ymO2OrQK9UyR3r+yNj1FTzhdwlKeNOCW8k1kpVZoJIaRGPoNflM+QDVcdg0V1zqF7vq23+uPsQ6OAzO83xMmVAdpggZaSOcepVpEfI8L3+WB8oZyRPYGqpAbgV7lALTkzj4cnQUByxR3Dk6ZTzkCzSEpOBzJuirV1ibrx2ESDffgEmQfjq9kZiyn4Zdf6uL9MuaQgqw7ODyKYcEQbALI+DDNCLtlnURoi9OCyQYZ0Y8d0APv2S0iYoetPJNtw3nvtCMqpVAahTditIrK035Rn3+ol6Hr4QRkitz3yv1dnWr2HueoCw4Gn+DJcutyR1btcq85XqGmtGn1MLf2luQ/oWSycM0KqVDUaPWhPT6bk3Ecafhvdrb7d+/5A0vBqreXHt3oEanvxskHGxHb2cJ7E4VQVvz5x03o9o7/+eaXWY5Pvex4MbYB27NQoteCjT/LSE1uLKYNJCflO0RGaDNIHGGqDM81f9On0EC7Qj2KoneRKM3EWkPKv0tmfam038g1H6U2kGqHbQa/svcplY/udr06oTtkmPzVerNrZNLeWy9iB6pHsxWb+noEitzZA5wUb2n64XDwnxzzShXDFZvF5m7p82CNA5GP0Apw4jvSrwb0as9jSqX5Iwe8MpbmPGc1hmj6SEvd4K4FtBIrreE2C8+/hwHKWLuJlGUCE1mApQs7FgDRjrPojNSi+xCo8wgFCY7Ox4fjz9HQjrchMfgsnBmzaMDS07ioN4xyjB+FdojVK1ZRvhYh6ixUPGZ8zG2fEcUsAAbWtDUO7D7ptQx20yrI9pKrdyaaGmighNBHlyn1okr63zbXPJtOZfevKs7H3F1GJQXpdCVRXD3Bxmt/Xg=)

**Comment by websitevirtuoso** [comment]
something like?

VTable: {
      VDialog: {
        VTextField: { variant: 'plain' },
      },
    },

**Comment by Winter979** [comment]
While that would do the trick, didnt need to do that before the latest release. 

So not sure if this is new intended feature, or a bug

**Comment by MetRonnie** [comment]
Here is a more simple repro I think: https://play.vuetifyjs.com/#eNqVkcFOwzAQRH9l5UsvJCmFXCJTCU58ARfCwcQLWHJsy16bQ5R/x3EbqCpUqTfPjvV2NPs6seCH5tG5OkVkHeOEo9OCcN8bAJ4qiR8iagqV8zYpiR66dfTQswlensh0MEESXglDHWxsJK0Myg3MMPesgApKOHcURQ7CS/hWkr4yaLfd/n49+u9kYNAiLHtGUbXZf0atLW+K94fKemGdsqUS2n5CN1qJukpCR8wU8hHPt1xMcVWO/5LkwSHK2kIerDUU87zd7PDm5AZZhsErRxCQotvz5iCzxeYbdlff17ctWx5tvWNvPwd2jhQ=

**Comment by johnleider** [comment]
This may have to do with Teleport.

**Comment by MetRonnie** [comment]
> This may have to do with Teleport.

Yes, I think so as I have found the global defaults do not apply to teleported content either
